packages/shared/src/getToken.ts

```diff
@@ -0,0 +1,160 @@
+import { inBrowser } from './browser';
+import type { ClerkStatus, GetTokenOptions, LoadedClerk } from './types';
+
+const POLL_INTERVAL_MS = 50;
+const MAX_POLL_RETRIES = 100; // 5 seconds of polling
+const TIMEOUT_MS = 10000; // 10 second absolute timeout
+
+type WindowClerk = LoadedClerk & {
+  status?: ClerkStatus;
+  loaded?: boolean;
+  on?: (event: 'status', handler: (status: ClerkStatus) => void, opts?: { notify?: boolean }) => void;
+  off?: (event: 'status', handler: (status: ClerkStatus) => void) => void;
+};
+
+function getWindowClerk(): WindowClerk | undefined {
+  if (inBrowser() && 'Clerk' in window) {
+    return (window as unknown as { Clerk?: WindowClerk }).Clerk;
+  }
+  return undefined;
+}
+
+class ClerkNotLoadedError extends Error {
+  constructor() {
+    super('Clerk: Timeout waiting for Clerk to load. Ensure ClerkProvider is mounted.');
+    this.name = 'ClerkNotLoadedError';
+  }
+}
+
+class ClerkNotAvailableError extends Error {
+  constructor() {
+    super('Clerk: getToken can only be used in browser environments.');
+    this.name = 'ClerkNotAvailableError';
+  }
+}
```

Comment:
We have a shared error type for runtime errors we should leverage here: https://github.com/clerk/javascript/blob/main/packages/shared/src/errors/clerkRuntimeError.ts

Each error should have a unique code